src/java/org/apache/cassandra/service/accord/txn/TxnNamedRead.java

```diff
@@ -130,9 +130,13 @@ public Future<Data> read(boolean isForWriteTxn, SafeCommandStore safeStore, Time
                  UnfilteredPartitionIterator partition = read.executeLocally(controller);
                  PartitionIterator iterator = UnfilteredPartitionIterators.filter(partition, read.nowInSec()))
             {
-                FilteredPartition filtered = FilteredPartition.create(PartitionIterators.getOnlyElement(iterator, read));
                 TxnData result = new TxnData();
-                result.put(name, filtered);
+                if (iterator.hasNext())
+                {
+                    FilteredPartition filtered = FilteredPartition.create(PartitionIterators.getOnlyElement(iterator, read));
```

Comment:
```suggestion
                    FilteredPartition filtered = FilteredPartition.create(iterator.next());
```
We should be able to simplify ^

The only other notable thing that `getOnlyElement()` seems to do is close the iterator, but `iterator` is already in a try-with-resources block above, so that shouldn't be an issue.